servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcUtils.java

```diff
@@ -263,10 +277,35 @@ static GrpcStatusException toGrpcException(Throwable cause) {
                 : new GrpcStatusException(toGrpcStatus(cause), () -> null);
     }
 
+    private static void validateStatusCode(HttpResponseStatus status) {
+        final int statusCode = status.code();
+        if (statusCode == OK.code()) {
+            return;
+        }
+        final GrpcStatusCode grpcStatusCode;
+        if (statusCode == BAD_GATEWAY.code() || statusCode == SERVICE_UNAVAILABLE.code() ||
+                statusCode == GATEWAY_TIMEOUT.code() || statusCode == TOO_MANY_REQUESTS.code()) {
+            grpcStatusCode = UNAVAILABLE;
+        } else if (statusCode == UNAUTHORIZED.code()) {
+            grpcStatusCode = UNAUTHENTICATED;
```

Comment:
I think `UNAUTHENTICATED` make sense in this case.

https://datatracker.ietf.org/doc/html/rfc7235#section-3.1
> The 401 (Unauthorized) status code indicates that the request has not
   been applied because it lacks valid authentication credentials for
   the target resource.

Where `403 Forbidden` means we authenticated the client, but permission was denied.

The unit test also confirms we use a consistent strategy to `grpc-java`.